### PR TITLE
Bump machine controller

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Update machine controller to a newer version.
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
None
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,41 +17,41 @@ require (
...
-	github.com/google/uuid v1.4.0
-	github.com/gophercloud/gophercloud v1.7.0
+	github.com/google/uuid v1.5.0
+	github.com/gophercloud/gophercloud v1.8.0
 	github.com/hetznercloud/hcloud-go v1.52.0
 	github.com/imdario/mergo v0.3.16
 	github.com/jackpal/gateway v1.0.10
 	github.com/kubermatic/grafanasdk v0.9.13
-	github.com/kubermatic/machine-controller v1.58.1-0.20231206121500-4eb097a46352
+	github.com/kubermatic/machine-controller v1.58.2
```
Comment (embik) [inline go.mod:46]: why bump this to a patch release and not the latest commit that is in `pkg/resources/machinecontroller/deployment.go`?

Comment (moadqassem) [reply go.mod:46]: Why not ?

Comment (embik) [reply go.mod:46]: Because we usually try to keep Go library and component image in sync so that API types match.

Comment (embik) [reply go.mod:46]: It's not a blocker to this PR; I was just curious.

Comment (moadqassem) [reply go.mod:46]: I see, no worries at all as that kinda makes sense. Will use the commit hash in the go mod file as well :)

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**